Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/non_rigid_mesh_registration.h

```diff
@@ -0,0 +1,773 @@
+* @param correspondences a vector given matching points between the `source` and the `target`
+*
+* \cgalNamedParamsBegin
+*   \cgalParamNBegin{number_of_iterations}
+*     \cgalParamDescription{the number of registration iterations using ICP, iterative closest point}
+*     \cgalParamType{unsigned int}
+*     \cgalParamDefault{`50`}
+*   \cgalParamNEnd
+*
+*   \cgalParamNBegin{point_to_plane_energy}
+*     \cgalParamDescription{the weight of the point to plane distance in the registration}
+*     \cgalParamType{double}
+*     \cgalParamDefault{`1`}
+*   \cgalParamNEnd
+*
+*   \cgalParamNBegin{point_to_point_energy}
+*     \cgalParamDescription{the weight of the point to matching point distance in the registration}
+*     \cgalParamType{double}
+*     \cgalParamDefault{`1`}
+*   \cgalParamNEnd
+*
+*   \cgalParamNBegin{as_rigid_as_possible_energy}
+*     \cgalParamDescription{defines the rigidity of the registration}
+*     \cgalParamType{double}
+*     \cgalParamDefault{`50`}
+*   \cgalParamNEnd
+*
+*   \cgalParamNBegin{maximum_matching_distance}
+*     \cgalParamDescription{the maximum distance for a vertex in `source` to match with a point in `target`. The default value 0 means no maximum matching distance.}
+*     \cgalParamType{double}
+*     \cgalParamDefault{`0`}
+*   \cgalParamNEnd
+*
+*   \cgalParamNBegin{vertex_point_map}
+*     \cgalParamDescription{a property map associating points to the vertices of `source`}
+*     \cgalParamType{a class model of `ReadablePropertyMap` with `boost::graph_traits<TriangleMesh>::%vertex_descriptor`
+*                    as key type and `%Point_3` as value type}
+*     \cgalParamDefault{`get_const_property_map(CGAL::vertex_point, source)`}
+*     \cgalParamExtra{If this parameter is omitted, an internal property map for `CGAL::vertex_point_t`
+*                     must be available in `TriangleMesh`.}
+*   \cgalParamNEnd
+*
+*   \cgalParamNBegin{geom_traits}
+*     \cgalParamDescription{an instance of a geometric traits class}
+*     \cgalParamType{a class model of `Kernel`}
+*     \cgalParamDefault{a \cgal Kernel deduced from the point type, using `CGAL::Kernel_traits`}
+*     \cgalParamExtra{The geometric traits class must be compatible with the vertex point type.}
+*   \cgalParamNEnd
+* \cgalNamedParamsEnd
+*/
+
+template <typename TriangleMesh,
+          typename PointRange,
+          typename VertexTranslationMap,
+          typename VertexRotationMap,
+          typename NamedParameters1 = parameters::Default_named_parameters,
+          typename NamedParameters2 = parameters::Default_named_parameters>
+void non_rigid_mesh_to_points_registration(TriangleMesh& source,
+  const PointRange& target,
+  VertexTranslationMap& vtm,
+  VertexRotationMap& vrm,
+  const std::vector<std::pair<typename boost::graph_traits<TriangleMesh>::vertex_descriptor, std::size_t>>& correspondences = std::vector<std::pair<typename boost::graph_traits<TriangleMesh>::vertex_descriptor, std::size_t>>(),
```

Comment:
Make it a range. This has also the advantage that we do not have to scroll to the right in the html.